packages/components/toast/stories/toast.stories.tsx

```diff
@@ -0,0 +1,154 @@
+import { Meta, StoryObj } from '@storybook/react'
+import { Button } from '@mochi-ui/button'
+import { Toast, ToastProps } from '../src'
+import { useToast } from '../src/hook/use-toast/use-toast'
+import { Toaster } from '../src/toaster'
+
+const meta: Meta<typeof Toast> = {
+  title: 'Components/Toast',
+  component: Toast,
+  parameters: {
+    layout: 'centered',
+  },
+  tags: ['autodocs'],
+  argTypes: {
+    type: {
+      control: 'select',
+      options: ['foreground', 'background'],
+    },
+    duration: {
+      control: 'number',
+    },
+    open: {
+      control: 'boolean',
+    },
+    forceMount: {
+      control: 'boolean',
+    },
+  },
+}
+
+const texts = [
+  'Lorem ipsum dolor sit amet, qui minim labore adipisicing minim sint cillum sint consectetur cupidatat.',
+  'Lorem ipsum dolor sit amet',
+  'Lorem ipsum dolor sit amet, qui minim labore adipisicing',
+]
+
+export default meta
+
+export const Default: StoryObj<typeof Toast> = {
+  render: function Render(props) {
+    const { toast } = useToast()
+    return (
+      <div className="w-[800px] min-h-[500px] max-h-[700px] overflow-y-auto flex items-center justify-center">
+        <div className="fixed top-0 right-0 max-w-[500px] mx-auto">
+          <Toaster />
+        </div>
+        <Button
+          onClick={() =>
+            toast({
+              title: texts[Math.floor(Math.random() * 3)],
+              description: texts[Math.floor(Math.random() * 3)],
+              ...props,
+            })
+          }
+        >
+          Toast
+        </Button>
+      </div>
+    )
+  },
+}
+
+export const StackToast: StoryObj<typeof Toast> = {
+  render: function Render() {
+    const { toast } = useToast()
+    return (
+      <div className="w-[800px] min-h-[500px] max-h-[700px] overflow-y-auto flex items-center justify-center">
+        <div className="fixed top-0 w-[600px] m-auto">
+          <Toaster />
+        </div>
+        <Button
+          onClick={() =>
+            toast({
+              layout: 'stack',
+              title: texts[Math.floor(Math.random() * 3)],
+              description: texts[Math.floor(Math.random() * 3)],
+              confirm: {
+                label: 'confirm',
+                altText: 'Confirm',
+                onClick: () => {
+                  alert('asod')
+                },
+              },
+              cancel: {
+                label: 'Cancel',
+                altText: 'Cancel',
```

Comment:
Can we demo the callback to close the toast after clicking Cancel?

![image](https://github.com/consolelabs/mochi-ui/assets/12707960/8b8c0806-1bc1-4758-adbf-e81b45f2371c)

Reply:
ToastAction close the toast onClick by default use `e.preventDefault` can solve this. But radix not recommend using toast to handle task that need user response, should use modal that style like a toast instead. However, i included a demo if a toast is used for task like this.

[screen-capture (10).webm](https://github.com/consolelabs/mochi-ui/assets/88917321/b217d6c6-6819-4005-859b-b8e24a72ce81)

https://www.radix-ui.com/primitives/docs/components/toast#action